test/modules/TestResultCoordinator/Reports/CountingReportGenerator.cs

```diff
@@ -136,7 +136,7 @@ public async Task<ITestResultReport> CreateReportAsync()
                 totalExpectCount,
                 totalMatchCount,
                 totalDuplicateResultCount,
-                unmatchedResults.AsReadOnly());
+                new List<TestOperationResult>(unmatchedResults).AsReadOnly());
```

Comment:
do we need to keep using List? is it ok to use queue in CountingReport? will this look the same in serialized json?

Reply:
We pass it in as a readonly list. We could instead pass the queue directly, but it wouldn't be readonly. We could find a readonly queue implementation, but also need to confirm that this would still serialize well.